### PR TITLE
Issue82 - Added scripts to program

### LINKED ISSUE #82: Add shell script to make program easier to run for end user
**Is your feature request related to a problem? Please describe.**
The customer relayed their frustrations with getting the program to run previously. This shell script should simplify the process and reduce the amount of clicks the user has to make in order to run the program. 

**Describe the solution you'd like**
The ideal solution would to make the program start in only 1 click. This is currently not possible, so a shell script is the best possible solution currently given the time constraints and budget limitations. A 1 click solution would require code signing, which costs money. 

**Describe alternatives you've considered**
I have considered making an Apple Script for the shell script, but problems would arise if the end user is on a different operating system than MacOS. 

**Additional context**
We hope in the future to enhance this feature to make running the program more user friendly.

### PR DESCRIPTION
**Pull request by Lyskawa-Aonyx**
Fixes #82 

**What was changed?**
Added shell script and AppleScript to root directory to enhance user friendliness of program startup.

*Here, describe what part of the application you changed (e.g. login page, database, etc.). If possible, mention what specific components were added, removed, or modified.*
Added esp.sh file to the program.
Added ESP.app to the program

**Why was it changed?**
It was changed based on customer feedback on their frustrations with getting the program to run before. 

*Here, describe the issue that you are fixing with this code, and why your code fixes it.*
I added the esp.sh program to make it easier for the user to run the program. I then made an AppleScript application that the user just needs to click on in order to run the program. The AppleScript relies on the esp.sh to function. 

*How to test the implemented changes?*
To test, you need to open terminal and navigate to the root folder of the program where esp.sh resides. You need to execute the command "chmod +x esp.sh" and then you will be able to run the script using "./esp.sh". If you want to test the AppleScript application, you need to double-click on ESP and follow the permissions it asks for. Docker does not need to be running before starting either script.

### CODE REVIEW

**Comment by SiriChandanaGarimella** [comment]
@Lyskawa-Aonyx - Great work! This adds great value to the client feedback. Here are some points to consider:

**Permissions**: Can you please review the permissions required? Many of the permissions listed may not be necessary for our specific use case. Please make sure that we request only the essential permissions to ensure user privacy.

**Create new issues for the following:**

**Application Termination**: We should create a new issue to develop a script that safely terminates the application and cleans up by removing all associated Docker images. This will ensure proper resource management and system cleanup.
**Cross-Platform Compatibility**: While the current implementation for macOS is suitable as the client is using a Mac laptop, we need to ensure future compatibility with Windows OS. We should create a new issue to enhance the script for cross-platform compatibility.

File: ESP.app/Contents/MacOS/applet
Comment (kate-holdener) [inline ESP.app/Contents/MacOS/applet:1]: This is showing up as a 'binary' file. Was this added accidentally?

File: ESP.app/Contents/Resources/Scripts/main.scpt
Comment (kate-holdener) [inline ESP.app/Contents/Resources/Scripts/main.scpt:1]: This is showing up as a binary file, was it added accidentally?

File: ESP.app/Contents/Resources/applet.icns
Comment (kate-holdener) [inline ESP.app/Contents/Resources/applet.icns:1]: This is showing up as a binary file, was it added accidentally?

**Comment by Lyskawa-Aonyx** [comment]
I have pushed a new version of ESP.app. I have implemented a preliminary termination behavior. I have also reviewed all of the app permissions and they are currently as minimal as I can make them. I have also fixed the sleep timer to instead check if the "docker compose up" command is finished before opening the user's browser.

File: ESP.app/Contents/Resources/applet.rsrc
[the remark this answers is not in the document]
Comment (Lyskawa-Aonyx) [reply ESP.app/Contents/Resources/applet.rsrc:1]: This was added intentionally for the end user (on Mac, I will be implementing a Windows version) to be able to easily open and close the program while it is in this 'beta' state. A much more elegant solution will be implemented for deployment.

File: ESP.app/Contents/MacOS/applet
Comment (Lyskawa-Aonyx) [reply ESP.app/Contents/MacOS/applet:1]: This was added intentionally for the end user (on Mac, I will be implementing a Windows version) to be able to easily open and close the program while it is in this 'beta' state. A much more elegant solution will be implemented for deployment.

File: esp.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+while ! docker  info > /dev/null 2>&1; do
+	echo "Waiting for Docker to start..."
+	sleep 5
+done
+
+#run command
+docker compose up & 
+
+#open app
+sleep 7
```
Comment (SiriChandanaGarimella) [inline esp.sh:22]: @Lyskawa-Aonyx - The comment says, 'I have also fixed the sleep timer to check if the "docker-compose up" command is finished before opening the user's browser.' However, I still see a sleep 7 in the code. Could you please verify if your latest changes have been pushed? If not, could you include this update as part of the stop script issue?

**Comment by SiriChandanaGarimella** [comment]
Closing this PR as this work is now included in the new PR #96

**Pull request closed without merging**